flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -869,6 +883,8 @@ private boolean triggerCheckpoint(
 					"invokable was not in state running.", checkpointMetaData.getCheckpointId(), getName(), e);
 				return false;
 			}
+		} finally {
+			FlinkUserSecurityManager.unmonitorSystemExitForCurrentThread();
 		}
```

Comment:
This is to protect against user-defined checkpoint function, but couldn't find a fast way to cover this in unit test (other than manual test). As user-defined checkpoint is unlikely to have exit function, I am fine to remove this instrumentation, so please let me know.